src/lib/sidenav/drawer.ts

```diff
@@ -273,6 +274,20 @@ export class MatDrawer implements AfterContentInit, AfterContentChecked, OnDestr
         this._restoreFocus();
       }
     });
+
+    /**
+     * Runs outside the zone the 'keydown' event listener. This avoids change detector to be fired
```

Comment:
How about:
```ts
/**
 * Listen to `keydown` events outside the zone so that change detection is not run every
 * time a key is pressed. Instead we re-enter the zone only if the `ESC` key is pressed
 * and we don't have close disabled.
 */
```

Reply:
Done!